packages/onboarding/src/utils/domain-suggester.ts

```diff
@@ -0,0 +1,108 @@
+const validDomains = [
+	'gmail.com',
+	'yahoo.com',
+	'hotmail.com',
+	'aol.com',
+	'outlook.com',
+	'comcast.net',
+	'icloud.com',
+	'msn.com',
+	'hotmail.co.uk',
+	'sbcglobal.net',
+];
+
+const extractDomainWithExtension = ( email: string ) => {
+	if ( email ) {
+		email = email.toLowerCase();
+		const atIndex = email.indexOf( '@' );
+		if ( atIndex !== -1 ) {
+			return email.slice( atIndex + 1 );
+		}
+	}
+	return;
+};
+
+const replaceEmailDomain = ( email: string, newDomain: string | null ) => {
+	if ( ! newDomain ) {
+		return newDomain;
+	}
```

Comment:
Let's not accept `null` here, since the value needed for this function to do its job. Whoever calls this function should check their values for null before calling it.

```suggestion
const replaceEmailDomain = ( email: string, newDomain: string ) => {
```